### DISCUSSION TITLE
Default Credit Card

### DISCUSSION THREAD

**Discussion by Ojay** [swellstores/community#32] [Help] [answered]
Hi all,

I'm just wondering if there's a way to set a credit card as `default` using Swell.js? It's easily done in the admin but I can't see how to do so via the api. In fact, when listing all of the cards using `await swell.account.listCards()` there's nothing to denote the default card.... assuming it's the first card returned in the array?

Any help on the above would be greatly appreciated.

Thanks!

Mark

![Example](https://uc6f59c5d109a5cdc6aa4d90a7d4.previews.dropboxusercontent.com/p/thumb/ABvF4cgrYIn78yvuyZbmdhsI_daSmrjjqF3e63pmQQfONhdh3DdkjxsnJlXByc83vCSd7lBI8_-3Ln9Kx_vO5nvsMfsi0PkL_k84BfbMKCpGhbsOoOWIGVBHDKkoAQtvPny4-slKdqbQtJDUPl4wLVIefmzn_8kfHOVJ1MJcS6mqhdZbQGdEUjoUGR3YB0Adn87l4GjFZSzNHn9NjNpOeQLjqftUR3DAedtFMWpeaYXMS-1UWTzokkCUkKvWKmpXJeRF89s0XFJNcEWQXV5NR2oX77Iaiyvf25o-ZTJd0730BgzQ0rBgudajJNU30omJ5SfAGG_OGlUQq5_sbuxDot7La-aIvrNspGSGJZucYJ9Hp-FY2sbdLUTJki6RxHji2jM/p.png)

**Comment by logeshswell** [comment, marked as answer]
@Ojay Yes definitely!
First, let me explain how to identify the default card.
```node
import swell from 'swell-js'
swell.init('my-store', 'public-key')

account = await swell.account.get()
```

`account.billing.account_card_id` gives the default card id from the list of cards that you fetched.

So, you could update this field using swell-js as follows:
```node
account = await swell.account.update({
  billing: {
    account_card_id: '<new_account_card_id>'
  }
});
```

**Reply by nikhil-varma** [reply]
@logeshswell Thanks for the help! One quick question.. how would you make a card as a default one? I don't see any documentation for this.. can you please redirect me to one if it exists? Also, looks like Swell allows deleting all the cards including the default one which is being used the an active subscription. Is that still the case? Or am I missing anything? Thanks in advance!

**Reply by logeshswell** [reply]
Documentation to update the default card is missing and our team is working on it now.

> how would you make a card as a default one? 

As I mentioned earlier, you could update the default card as follows:
```node
account = await swell.account.update({
  billing: {
    account_card_id: '<new_account_card_id>'
  }
});
```

Does Swell support deleting default card? Yes
Does Swell support deleting a card that is being used in an active subscription? Yes, but it will still remain associated to the subscription unless it is modified.

**Reply by nikhil-varma** [reply]
@logeshswell Thanks again! 
> Does Swell support deleting a card that is being used in an active subscription? Yes, but it will still remain associated to the subscription unless it is modified.

If I understand it correct, if the card is deleted, the subscription will continue to deduct amount on the last associated card?

**Reply by ericingram** [reply]
It will only auto-charge the current account default card. If there is no current default (perhaps because it was deleted and another one wasn't set as default), then it will have an error charging, but as soon as the default is set to another card, any existing invoice will automatically be attempted using that new default.

**Reply by woutercouvaras** [reply]
Hi, this thread has been helpful, but i'm struggling to set a specfic card on a subscription.  If a user has multiple cards, I want ot be able to specify the card that should be charged.

I've tried a variety of things.  I've tried setting the `account_card_id`, I've tried setting these basic values:
```
billing: {
  token: xxx,
  exp_month: xxx,
  exp_year: xxx,
  brand: xxx,
  last4: xxx
}
```

but the `billing.card` property always shows as `null` when viewing the subscription record in the console.  Any ideas?

**Reply by ericingram** [reply]
It should be like this
```
billing: {
  card: {
    token: xxx,
    exp_month: xxx,
    exp_year: xxx,
    brand: xxx,
    last4: xxx
  }
}
```

**Reply by woutercouvaras** [reply]
Sorry, there was a typo in my example.  I do have the card property in there:
```
billing: {
  method: 'paystack',
  card: { ...card } // I've done this in different ways too
}
```

here's an example of the data contained in teh card variable (as printed out during a transaction):
```
{
  token: 'xxx redacted xxx',
  exp_month: 12,
  exp_year: 2030,
  brand: 'visa',
  last4: '4081'
}
```

but if I check the subscription in swell (using the console), the card is still `null`:
```
"billing": {
    "method": "paystack",
    "card": null,
    "use_account": false
  },
```

**Reply by ericingram** [reply]
`method` should be `card` instead of paystack

**Reply by woutercouvaras** [reply]
That worked, thank you.  I have a custom card payment method setup, so I made a poor assumption here.  I've just checked the docs and they do indicate the available options, my apologies!  Thanks for helping!
